trigger/index.ts

```diff
@@ -0,0 +1,88 @@
+// @ts-ignore
+import useSWR, {
+  Middleware,
+  SWRHook,
+  Key,
+  Fetcher,
+  SWRConfiguration,
+  SWRResponse
+} from 'swr'
+import { useCallback } from 'react'
+
+import { withMiddleware } from '../src/utils/with-middleware'
+import { serialize } from '../src/utils/serialize'
+
+export const trigger = ((<Data, Error>(useSWRNext: SWRHook) => (
+  key: Key,
+  fetcher: Fetcher<Data>,
+  config: SWRConfiguration<Data, Error>
+): SWRTriggerResponse<Data, Error> => {
+  // Disable all revalidations.
+  config.revalidateOnFocus = false
+  config.revalidateWhenStale = false
+  config.revalidateOnReconnect = false
+
+  // Disable options that are imcompatible with trigger mode.
+  config.suspense = false
+  config.shouldRetryOnError = false
+  config.refreshInterval = 0
+  config.loadingTimeout = 0
+  config.fetcher = undefined
+
+  // Extract callbacks.
+  const { onSuccess, onError } = config
+
+  // Disable fetcher.
+  const swr = useSWRNext(key, null, config) as SWRTriggerResponse
+
+  // Serialize the key and get args.
+  const [keyStr, args] = serialize(key)
+
+  // Method to trigger the request manually.
+  swr.trigger = useCallback(
+    (...extraArgs) => {
+      // If it's not ready to fetch, we throw an error.
+      if (!keyStr) throw new Error('Empty SWR key.')
+      if (!fetcher) throw new Error('Missing SWR fetcher.')
+
+      // Mutate the SWR data and return the result.
+      return swr.mutate(
```

Comment:
RE: 

> Sometimes, mutation APIs (POST, PUT, DELETE) return different payload from the GET API. It might be an issue if we write back to the cache with the same key. This behavior needs to be reconsidered.

When the user is not interested in updating the global cache for `key` they can use the `fetcher` right away i.e. they won't need `useSWRTrigger`.

Reply:
Yes. But I'm also thinking about an option to prevent updating the cache, but then `useSWRTrigger` will have nothing "SWR". Haven’t decided yet.